In [2]:
# Data visualization and exploratory data analysis
# %pip install matplotlib
import pandas as pd
# Data preparation and EDA tools
from dataprep.datasets import load_dataset  
from dataprep.eda import create_report

# venv310:
#from ydata_profiling import ProfileReport
#from autoviz.AutoViz_Class import AutoViz_Class
#import sweetviz as sv
#from phik import resources, report
#import numpy as np

ModuleNotFoundError: No module named 'dataprep'

<h1 style="
    font-family: 'Arial', sans-serif;
    font-size: 2.5em;
    background: linear-gradient(90deg, #FF7F50, #1E90FF);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    text-align: center;
    text-shadow: 2px 2px 5px rgba(0, 0, 0, 0.3);
    margin-bottom: 20px;
    padding: 10px;
">
  Loading and  Exploring the Datasets with DataPrep
</h1

In [8]:
# Load the dataset from the dataprep library , classifiction dataset
df = pd.read_csv('datasets/codon_usage.csv')
df.head()

C:\Users\Home\AppData\Local\Temp\ipykernel_11584\130701772.py:2: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('datasets/codon_usage.csv')


Kingdom  DNAtype  SpeciesID  Ncodons  \
0     vrl        0     100217     1995   
1     vrl        0     100220     1474   
2     vrl        0     100755     4862   
3     vrl        0     100880     1915   
4     vrl        0     100887    22831   

                               SpeciesName      UUU      UUC      UUA  \
0  Epizootic haematopoietic necrosis virus  0.01654  0.01203  0.00050   
1                         Bohle iridovirus  0.02714  0.01357  0.00068   
2             Sweet potato leaf curl virus  0.01974   0.0218  0.01357   
3             Northern cereal mosaic virus  0.01775  0.02245  0.01619   
4           Soil-borne cereal mosaic virus  0.02816  0.01371  0.00767   

       UUG      CUU  ...      CGG      AGA      AGG      GAU      GAC  \
0  0.00351  0.01203  ...  0.00451  0.01303  0.03559  0.01003  0.04612   
1  0.00678  0.00407  ...  0.00136  0.01696  0.03596  0.01221  0.04545   
2  0.01543  0.00782  ...  0.00596  0.01974  0.02489  0.03126  0.02036   
3  0.00992  0.01567  ...  0.00366  0.01410  0.01671  0.03760  0.01932   
4  0.03679  0.01380  ...  0.00604  0.01494  0.01734  0.04148  0.02483   

       GAA      GAG      UAA      UAG      UGA  
0  0.01203  0.04361  0.00251  0.00050  0.00000  
1  0.01560  0.04410  0.00271  0.00068  0.00000  
2  0.02242  0.02468  0.00391  0.00000  0.00144  
3  0.03029  0.03446  0.00261  0.00157  0.00000  
4  0.03359  0.03679  0.00000  0.00044  0.00131  

[5 rows x 69 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13028 entries, 0 to 13027
Data columns (total 69 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Kingdom      13028 non-null  object 
 1   DNAtype      13028 non-null  int64  
 2   SpeciesID    13028 non-null  int64  
 3   Ncodons      13028 non-null  int64  
 4   SpeciesName  13028 non-null  object 
 5   UUU          13028 non-null  object 
 6   UUC          13028 non-null  object 
 7   UUA          13028 non-null  float64
 8   UUG          13028 non-null  float64
 9   CUU          13028 non-null  float64
 10  CUC          13028 non-null  float64
 11  CUA          13028 non-null  float64
 12  CUG          13028 non-null  float64
 13  AUU          13028 non-null  float64
 14  AUC          13028 non-null  float64
 15  AUA          13028 non-null  float64
 16  AUG          13028 non-null  float64
 17  GUU          13028 non-null  float64
 18  GUC          13028 non-null  float64
 19  GUA 

In [10]:

# Create a report
report = create_report(df)
# Display the report
report.show_browser()

# save the report to a file
report.save("codon_usage_report.html")
# Display the report on the VS Code

NameError: name 'create_report' is not defined

## <span style="color: #20C997;">Personal Analysis</span>
I picked up  codon usage frequencies in the genomic coding DNA dataset, as classification dataset, It has 69 features and 13028 rows with no missing cells and duplicates rows. 65 are numerical columns and 4 are categorical columns.  DataPred detected 25 similar distribution and 25 Skewed, one Constant Length, 3 High Cardinality  and 11 Zeros.